_posts/languages/nodejs/2000-01-01-yarn-2.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Node.js
+nav: Yarn 2 and Yarn 3 Compatibility
+modified_at: 2021-12-10 00:00:00
+tags: nodejs
+---
+
+# What's new in new Yarn major version
+
+Yarn was developed as an alternative way to handle Node.JS dependencies to NPM
+but was fully compatible with it. They were both manipulating the
+`node_modules` directory and switching to one or the other tool was
+straightforward.
+
+Yarn 2 and Yarn 3 have evolved to use another approached, aiming at making
+dependency management for Node.JS application more efficient, especially during
+its deployment.
+
+{% warning %}
+This article only covers application using the Node.JS buildpack and not other
+buildpacks which include installation of a Node.JS stack like Ruby or PHP. If
+Yarn 2 or 3 is used alongside a Ruby or Node.JS application, the
+[multi-buildpacks]({% post_url platform/deployment/buildpacks/2000-01-01-multi
+%}) should be used.
+{% endwarning %}
+
+You are not force to migrated to Yarn 2 or 3, compatibility with Yarn 1 will be
+kept as long as possible, but to enjoy fastest deployments and last patches,
+you are encouraged to upgrade your project or get back to NPM.
+
+New versions of Yarn have 2 ways to work:
+
+* With Plug'n Play: it is expected to checkout the application dependencies
+  `.yarn/cache` in your git repository. It will accelerate deployment since it
+  won't be necessary to download dependencies they are already in your repository.
+  Don't worry yarn is optimizing and compressing them to prevent using too much space.
```

Comment:
Just for consistency as you spelled it that way in the page:
```suggestion
  Don't worry Yarn is optimizing and compressing them to prevent using too much space.
```